lib/TPP/ToBlockLayoutAndBack.cpp

```diff
@@ -444,7 +444,7 @@ mlir::linalgx::packVNNIMatmulOp(RewriterBase &rewriter,
 
   Location loc = matmulOp.getLoc();
   auto blockingFactor =
-      vnni::utils::getVNNIBlockingFactor(matmulOp.getInputs()[1].getType());
+      vnni::utils::getVnniBlockingFactor(matmulOp.getInputs()[1].getType());
```

Comment:
We also need to de-duplicate `packMatmulOpImpl` and `packVNNIMatmulOp`

Reply:
The problem here is that `packVNNIMatmulOp` assume already a packed matmul. Currently VNNI requires first a "traditional" pack.

Reply:
I was looking at OneDNN (sc_data_format.hpp) and basically with @adam-smnk we were discussing something along the line:
```
func.func @gemm(%arg0: tensor<>, %arg1: tensor<>, %arg2: tensor<>) -> tensor<> {
    %i = arith.constant 32 : index
    %k = arith.constant 32 : index
    // We cannot use only attributes as the blocking factors can be values.
    %0 = layout.reorder %arg0 [KCSck2c, [%c, %k]] : tensor<256x512> -> tensor<256x512, #KCSck2c> // 2 packs
    %1 = layout.reorder %arg1 [KCSck2c, [32, 32]] : tensor<256x512> -> tensor<256x512, #KCSck2c>
    %2 = layout.reorder %arg2 [KCSck2c, [32, 32]] : tensor<256x512> -> tensor<256x512, #KCSck2c>
    /*** some ops ***/
    %4 = layout.reorder %3 [KCSck2c, [32, 32]] : tensor<256x512, #KCSck2c> -> tensor<256x512>
    return %4: tensor<256x512>
}
```
This would allow the compiler to have an analysis pass to decide the optional layout for each operation and simplify the packing for VNNI, and company.